transport.go

```diff
@@ -202,6 +224,56 @@ func (t *transport) Dial(ctx context.Context, raddr ma.Multiaddr, p peer.ID) (tp
 	return conn, nil
 }
 
+func (t *transport) holePunch(ctx context.Context, p peer.ID, network string, addr *net.UDPAddr) error {
+	pconn, err := t.connManager.Dial(network, addr)
```

Comment:
@vyzo Where is the code to stop sending packets when it sees a connection on the listener ?

Reply:
it's in the listener -- see `setupConn`.